dashboard/client/src/service/task.ts

```diff
@@ -2,10 +2,18 @@ import { StateApiResponse } from "../type/stateApi";
 import { Task } from "../type/task";
 import { get } from "./requestHandlers";
 
-export const getTasks = () => {
-  return get<StateApiResponse<Task>>("api/v0/tasks?detail=1&limit=10000");
+export const getTasks = (jobId: string | null) => {
+  let url = "api/v0/tasks?detail=1&limit=10000";
+  if (jobId) {
+    url += `&job_id=${jobId}`;
+  }
+  return get<StateApiResponse<Task>>(url);
 };
 
-export const downloadTaskTimelineHref = (jobId: string) => {
-  return `/api/v0/tasks/timeline?job_id=${jobId}&download=1`;
+export const downloadTaskTimelineHref = (jobId: string | null) => {
```

Comment:
why is `jobId` optional?

Reply:
I think it could be useful for things like serve in the future. I just did it because it is a simple change. Doesn't have to be included in this PR if you don't want to